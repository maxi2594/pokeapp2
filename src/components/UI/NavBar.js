import React, { useContext } from 'react'
import { Link, NavLink, useHistory } from 'react-router-dom'
import { pokeContext } from '../../context/pokeContext'

export const Navbar = () => {

    const { handleSetUrL }=useContext(pokeContext)

    const handleInico = () => {
        handleSetUrL(`https://pokeapi.co/api/v2/pokemon-form/?limit=5&offset=5`);
    }


    return (
        <nav className="container-fluid navbar navbar-expand-sm navbar-dark bg-dark">
            
            <Link 
                className="navbar-brand" 
                to="/"
            >
                Asociaciones
            </Link>

            <div className="navbar-collapse">
                <div className="navbar-nav">

                    <NavLink 
                        activeClassName="active"
                        className="nav-item nav-link" 
                        exact
                        to="/pokes"
                        onClick={ handleInico }
                    >
                        Inicio
                    </NavLink>

                    <NavLink 
                        activeClassName="active"
                        className="nav-item nav-link" 
                        exact
                        to="/dc"
                    >
                        DC
                    </NavLink>

                    <NavLink 
                        activeClassName="active"
                        className="nav-item nav-link" 
                        exact
                        to="/search"
                    >
                        Search
                    </NavLink>
                </div>
            </div>
        </nav>
    )
}