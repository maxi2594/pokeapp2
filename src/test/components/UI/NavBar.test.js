import { mount, shallow } from "enzyme"
import toJson from "enzyme-to-json";
import { MemoryRouter } from "react-router";

import { Navbar } from "../../../components/UI/NavBar"
import { pokeContext } from "../../../context/pokeContext";

describe('Prubas en <NavBar/>', () => {

    const handleSetUrL = jest.fn()

    const wrapper = shallow(
        <pokeContext.Provider value={{handleSetUrL}}>
                <Navbar/>

        </pokeContext.Provider>
    )

    test('should render properly', () => {
        
        expect(toJson( wrapper)).toMatchSnapshot();
    })
    
    test('should include navlink-inicio', () => {
        
        const wrapper = mount(
            <pokeContext.Provider value={{handleSetUrL}}>
                <MemoryRouter>
                    <Navbar/>
                </MemoryRouter>

            </pokeContext.Provider>
        )

        expect( wrapper.find('NavLink').exists()).toBe(true)
        expect( wrapper.find('NavLink').at(0).text()).toBe('Inicio')

    })

    test('should handleSetUrL be colled on click', () => {

        const wrapper = mount(
            <pokeContext.Provider value={{handleSetUrL}}>
                <MemoryRouter>
                    <Navbar/>
                </MemoryRouter>

            </pokeContext.Provider>
        )

        const inicioNav = wrapper.find('NavLink').at(0);
        inicioNav.simulate('click')
        
        expect(handleSetUrL).toBeCalledTimes(1);    
    })
    
    
    
    
})
