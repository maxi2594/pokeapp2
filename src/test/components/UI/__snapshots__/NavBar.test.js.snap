// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prubas en <NavBar/> should render properly 1`] = `<Navbar />`;
